sklearn/preprocessing/_encoders.py

```diff
@@ -398,9 +401,25 @@ def transform(self, X):
             # categories that occur after them to avoid an empty column.
 
             keep_cells = X_int != to_drop
+
+            if isinstance(self.drop, str) and self.drop == 'if_binary':
+                keep_cells = keep_cells.T
+                X_int = X_int.T
+                for i in range(n_features):
+                    if len(self.categories_[i]) > 2:
+                        keep_cells[i] = True
+                    else:
+                        X_int[i][X_int[i] > 0] -= 1
```

Comment:
If X_int was not transposed, `X_int[X_int[i] > 0, i] -= 1` should work.

Reply:
That raises this error `boolean index did not match indexed array along dimension 0; dimension is 3 but corresponding boolean dimension is 5`.
Using `X_int[:, col][X_int[:, col] > 0] -= 1` instead.

Reply:
you need to put the masking in the first indexing `[..., ...]`